lib/chef/knife/bootstrap.rb

```diff
@@ -588,13 +598,30 @@ def register_client
       def perform_bootstrap(remote_bootstrap_script_path)
         ui.info("Bootstrapping #{ui.color(server_name, :bold)}")
         cmd = bootstrap_command(remote_bootstrap_script_path)
+        bootstrap_run_command(cmd)
+      end
+
+      # Actual bootstrap command to be run on the node.
+      # Handles recursive calls if su USER failed to authenticate.
+      def bootstrap_run_command(cmd)
         r = connection.run_command(cmd) do |data|
           ui.msg("#{ui.color(" [#{connection.hostname}]", :cyan)} #{data}")
         end
         if r.exit_status != 0
-          ui.error("The following error occurred on #{server_name}:")
-          ui.error(r.stderr)
-          exit 1
+          stderr = (r.stderr + r.stdout).strip
+
+          if stderr.match?("su: Authentication failure")
```

Comment:
This really feels like something that we should be handling on the train side. We need to either entirely handle this sort of error handling within chef or entirely abstract it with train. For this use case it feels like abstracting it in train and making it something we can reuse in inspect/ohai/chef is worth the effort.

Reply:
Added custom user-defined error here https://github.com/inspec/train/pull/636/files#diff-d143a9dd9f1352e6173b1112bacec369R251-R253, and raise `Train::UserError` with reason `bad_su_user_password`